packages/flutter_tools/lib/src/android/android_emulator.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+import '../android/android_sdk.dart';
+import '../android/android_workflow.dart';
+import '../base/file_system.dart';
+import '../base/process.dart';
+import '../emulator.dart';
+import '../globals.dart';
+import 'android_sdk.dart';
+
+class AndroidEmulators extends EmulatorDiscovery {
+  @override
+  bool get supportsPlatform => true;
+
+  @override
+  bool get canListAnything => androidWorkflow.canListEmulators;
+
+  @override
+  Future<List<Emulator>> get emulators async => getEmulatorAvds();
+}
+
+class AndroidEmulator extends Emulator {
+  AndroidEmulator(String id, [this._properties])
+      : super(id, _properties != null && _properties.isNotEmpty);
+
+  Map<String, String> _properties;
+
+  @override
+  String get name => _properties['hw.device.name'];
+
+  @override
+  String get manufacturer => _properties['hw.device.manufacturer'];
+
+  @override
+  String get label => _properties['avd.ini.displayname'];
+
+  @override
+  Future<bool> launch() async {
+    final RunResult launchResult =
+        await runAsync(<String>[getEmulatorPath(), '-avd', id]);
+    if (launchResult.exitCode != 0) {
+      printError(
+          'Error: emulator exited with exit code ${launchResult.exitCode}');
```

Comment:
It may not be necessary to print this explicitly - the `printError('$launchResult')` will print the stdout+stderr.

Reply:
Yep, message seems fine without the first line so I've removed it (the original code was copied from `android_device.dart` for ADB install, but it seems like that's the only place doing that!).